core/lib/zksync_core/src/consistency_checker/mod.rs

```diff
@@ -504,11 +504,20 @@ impl ConsistencyChecker {
             .0
             .saturating_sub(self.max_batches_to_recheck)
             .into();
+
+        let last_processed_batch = self
+            .pool
+            .connection()
+            .await?
+            .blocks_dal()
+            .get_consistency_checker_last_processed_l1_batch()
+            .await?;
+
         // We shouldn't check batches not present in the storage, and skip the genesis batch since
         // it's not committed on L1.
         let first_batch_to_check = first_batch_to_check
             .max(earliest_l1_batch_number)
-            .max(L1BatchNumber(1));
+            .max(L1BatchNumber(last_processed_batch.0 + 1));
```

Comment:
Nit: You can add `u32` to `L1BatchNumber` directly, so this can be simplified to `.max(last_processed_batch + 1)`.

Reply:
almost directly, I'd still have to use `*last_processed_batch + 1` so I think it's not worth the hassle